File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    dic = {i:v for i,v in enumerate(sequence)}
    answer = []
    start, sum = 0,0

    for i,v in enumerate(sequence):
        end = i
        sum += v
        while sum > k :
            sum -= dic[start]
            start += 1
        if sum == k: answer.append([start,end])

    answer.sort(key = lambda x: (x[1]-x[0],x[0]))
    return answer[0]

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    let answer = null;
    let sum = sequence[0];
    let s = 0, e = 0;

    while (s < sequence.length && e < sequence.length) {
        if (sum === k) {
            if (answer == null || e - s < answer[1] - answer[0]) {
                answer = [s, e];
            }
        }
        if (sum < k || s == e) sum += sequence[++e];
        else sum -= sequence[s++];
    }

    return answer;
}